modules/ocf_hpc/manifests/compute.pp

```diff
@@ -0,0 +1,34 @@
+class ocf_hpc::compute {
+  require ocf_hpc
+
+  # install proprietary nvidia drivers
+  package { ['nvidia-driver', 'nvidia-settings', 'nvidia-cuda-toolkit']:; }
+
+  file { '/etc/slurm-llnl/gres.conf':
+    content => template('ocf_hpc/gres.conf.erb'),
+    mode    => '0644',
+    owner   => 'slurm',
+    group   => 'slurm',
+  }
+  # Kernel option to enable memory as a consumable gres resource
+  augeas { 'grub':
+    incl    => '/etc/default/grub',
+    lens    => 'ShellVars.lns',
+    changes => [
+      "set GRUB_CMDLINE_LINUX '\"cgroup_enable=memory swapaccount=1\"'",
+    ],
+  } -> exec { 'update-grub':
```

Comment:
This will cause `update-grub` to run every time Puppet runs, which probably isn't what you want.

Try instead setting the `refreshonly => true` parameter on the `update-grub` resource and changing the `->` to a `~>`. This way, `update-grub` will only run when augeas actually modifies grub's configuration file.